### PR TITLE
WIP: Update to php72

### PR DESCRIPTION
**Pull request by basz**
comments welcome

- php72+
- utilize ~laminas-coding-standard~ PSR12

note: ~currently waiting on pending doctrine-orm-module update~ apparently it works...

should update sim-queue now

### CODE REVIEW

**Comment by basz** [comment]
@roelvanduijnhoven what do you think we should do?

release a *last* minor version and mark it as EOL with just the laminas changes.

then release a new major version targeting php72+ and maybe containing other updates/changes?

This could/should be done for slm-queue too then. I can take point on that...

**Comment by roelvanduijnhoven** [comment]
@basz I would feel fine just releasing a new major version. Rationele: people that migrate to Laminas will probably actively develop on their app, and have no problem meeting the 7.2+ requirement.

**Comment by roelvanduijnhoven** [comment]
@basz Had a look to the PR. Looks great to me! 

Let's also backport the PSR12 formatting to SlmQueue :).

Your latest PR breaks the test suite. I think you cannot type hint some attributes, because that will break some interfaces in Laminas / SlmQueue.

**Comment by basz** [comment]
Yes, test will fail until sim-queue has been updated. On that now ;-)

**Comment by basz** [comment]
@roelvanduijnhoven now that sim-queue is merged are you able to restart travis for this PR? Otherwise i'll make a dummy commit... thx!

**Comment by roelvanduijnhoven** [comment]
@basz Restarted. Also increased your contributor level on GitHub to maintainer: so maybe now you have the superpower yourself ;)?

**Comment by basz** [comment]
would you do a final review please?

File: composer.json [hunk 1]
```diff
@@ -18,14 +18,19 @@
...
+        {
+            "name": "Bas Kamer",
+            "email": "baskamer@gmail.com",
+            "homepage": "https://bushbaby.nl"
         }
     ],
     "require": {
-        "php": "^5.5 || ^7.0",
+        "php": "^7.2",
         "laminas/laminas-eventmanager": "^2.6 || ^3.0",
```
Comment (roelvanduijnhoven) [inline composer.json]: Have a look at SlmQueue. I raised all depencies to the minimum version I could find that actually support PHP 7.2. I did that also for the dev-dependencies.

Little trick I used to do this fast is to look at the HTML, and search for `PHP 7.`. For example in view-source:https://github.com/laminas/laminas-eventmanager/releases.

File: src/Factory/DoctrineQueueFactory.php [hunk 2]
```diff
@@ -28,16 +28,17 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
...
-        return $queue;
+        return new DoctrineQueue($connection, $queueOptions, $requestedName, $jobPluginManager);
     }
 
     /**
      * {@inheritDoc}
      */
-    public function createService(ServiceLocatorInterface $serviceLocator, $name = '', $requestedName = '')
-    {
+    public function createService(
```
Comment (roelvanduijnhoven) [inline src/Factory/DoctrineQueueFactory.php]: Once you raise your dependencies to a level that is PHP 7.2 compatible I think you will find that these no longer this backwards compatibility layer. I would advise to migrate to the new style, and remove the old `createService` functions.

File: src/Queue/DoctrineQueue.php [hunk 3]
```diff
@@ -270,54 +270,48 @@ public function bury(JobInterface $job, array $options = [])
...
+    public function peek(int $id): JobInterface
     {
-        $sql  = 'SELECT * FROM ' . $this->options->getTableName().' WHERE id = ?';
-        $row  = $this->connection->fetchAssoc($sql, [$id], [Type::SMALLINT]);
+        $sql  = 'SELECT * FROM ' . $this->options->getTableName() . ' WHERE id = ?';
+        $row  = $this->connection->fetchAssoc($sql, [$id], [Types::SMALLINT]);
 
-        if (!$row) {
-            throw new Exception\JobNotFoundException(sprintf("Job with id '%s' does not exists.", $id));
+        if (! $row) {
```
Comment (roelvanduijnhoven) [inline src/Queue/DoctrineQueue.php]: Not sure... Is this PSR-12? Looks weird to me with the extra space.

File: src/Queue/DoctrineQueue.php [hunk 4]
```diff
@@ -383,7 +373,7 @@ public function release(JobInterface $job, array $options = [])
      * @param $options array
      * @return DateTime
      */
-    protected function parseOptionsToDateTime($options)
+    protected function parseOptionsToDateTime($options): DateTime
```
Comment (roelvanduijnhoven) [inline src/Queue/DoctrineQueue.php]: `array?` And remove docblock?

File: src/Queue/DoctrineQueue.php [hunk 5]
```diff
@@ -417,7 +407,7 @@ protected function parseOptionsToDateTime($options)
                     try {
                         // first try ISO 8601 duration specification
                         $delay = new DateInterval($options['delay']);
-                    } catch (\Exception $e) {
+                    } catch (Exception $e) {
```
Comment (roelvanduijnhoven) [inline src/Queue/DoctrineQueue.php]: Personally I would rather keep the `\`, so I can see that this one is from the standard library. But probably boils down to taste.

File: src/Strategy/IdleNapStrategy.php [hunk 6]
```diff
@@ -17,26 +17,20 @@ class IdleNapStrategy extends AbstractStrategy
      */
     protected $napDuration = 1;
 
-    /**
-     * @param int $napDuration
-     */
-    public function setNapDuration($napDuration)
+    public function setNapDuration(int $napDuration)
```
Comment (roelvanduijnhoven) [inline src/Strategy/IdleNapStrategy.php]: `return void`

**Pull request merged**